File: _BHCVRP_Python_Version/BHCVRP_Python_Version/cujae_inf_citi_om/HFVRP.py
import sys
import os
from tools.LoadFile import LoadFile
from generator.controller.StrategyHeuristic import StrategyHeuristic
from data.ProblemType import ProblemType
from tools.OrderType import OrderType
from factory.interfaces.HeuristicType import HeuristicType
from data.Problem import Problem


def main():
    try:
        file_output = open("D:\\Escuela\\BHCVRP_Python_Version\\Resultados\\HFVRP\\Instancia_HFVRP_C1_6_1\\Resultado_C1_6_1_K1.txt", "w")
        sys.stdout = file_output

        path_files = "D:\\Escuela\\BHCVRP_Python_Version\\Resultados\\HFVRP\\C1_6_1.txt"

        load_file = LoadFile()

        load_file.load_file(path_files)  # i + 1

        id_customers = []
        axis_x_customers = []
        axis_y_customers = []
        request_customers = []

        id_depots = []
        axis_x_depots = []
        axis_y_depots = []
        count_vehicles = []
        capacity_vehicles = []

        list_distances = []
        problem = Problem.get_problem()
        load_file.load_count_vehicles_for_depot(count_vehicles)
        load_file.load_capacity_vehicles_for_hfvrp(capacity_vehicles)
        load_file.is_load_customers(id_customers, axis_x_customers, axis_y_customers, request_customers)
        load_file.is_load_depots(id_depots, axis_x_depots, axis_y_depots)

        load_file.fill_list_distances(id_customers, axis_x_customers, axis_y_customers, id_depots, axis_x_depots,
                                      axis_y_depots, list_distances)

        type_problem = ProblemType.HFVRP
        order_type = OrderType.Descending
        heuristic_type = HeuristicType.KilbyAlgorithm

        if StrategyHeuristic.get_strategy_heuristic().load_hfvrp(id_customers, request_customers, id_depots,
                                                                 count_vehicles[0], capacity_vehicles[0],
                                                                 list_distances,
                                                                 axis_x_customers, axis_y_customers, axis_x_depots,
                                                                 axis_y_depots, type_problem, order_type):

            StrategyHeuristic.get_strategy_heuristic().execute_heuristic(1, heuristic_type)
            result = StrategyHeuristic.get_strategy_heuristic().get_best_solution()
            cost = StrategyHeuristic.get_strategy_heuristic().get_total_cost_solution()
            request_by_route = len(StrategyHeuristic.get_strategy_heuristic().get_request_by_route())
            time = StrategyHeuristic.get_strategy_heuristic().get_time_execute()

            print(" ")
            print("------------------------------------------")
            print("CANTIDAD DE EJECUCIONES: 100")
            print("HEURÍSTICA DE CONSTRUCCIÓN: " + str(heuristic_type))
            print("COSTO TOTAL: " + str(cost))
            print("TOTAL DE RUTAS: " + str(request_by_route))
            print("TIEMPO DE EJECUCIÓN: " + str(time) + " milisegundos")
            print(" ")

            for j in range(request_by_route):
                print("R" + str(j + 1) + str(result.get_list_routes()[j].get_list_id_customers()))

            print("------------------------------------------")

        file_output.close()
        sys.stdout = sys.__stdout__

    except IOError as e:
        print(e)


if __name__ == "__main__":
    main()

File: BHCVRP_Java_Version/src/test/HFVRP.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package test;

import cujae.inf.citi.om.data.ProblemType;
import cujae.inf.citi.om.factory.interfaces.HeuristicType;
import cujae.inf.citi.om.generator.controller.StrategyHeuristic;
import cujae.inf.citi.om.generator.solution.Solution;
import cujae.inf.citi.om.tools.OrderType;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.PrintStream;
import java.lang.reflect.InvocationTargetException;
import java.util.ArrayList;

/**
 *
 * @author kmych
 */
public class HFVRP {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) throws IOException, IllegalArgumentException, SecurityException, ClassNotFoundException, InstantiationException, IllegalAccessException, InvocationTargetException, NoSuchMethodException 
    {
        try {
            FileOutputStream fileOutputStream = new FileOutputStream("/D:/Escuela/BHCVRP/ResultadosHFVRP/Instancia_HFVRP_p14/Resultado_NN20.txt");
            PrintStream printStream = new PrintStream(fileOutputStream);

            System.setOut(printStream);

            String pathFiles = "modified-hfvrp//HFVRP_p14";
                //int totalInstances = 5;
                LoadFile loadFile = new LoadFile();

                //for(int i = 1; i < totalInstances; i++)
                //{
                    loadFile.loadFile(pathFiles); //i + 1

                    ArrayList<Integer> idCustomers = new ArrayList<Integer>();
                    ArrayList<Double> axisXCustomers = new ArrayList<Double>();
                    ArrayList<Double> axisYCustomers = new ArrayList<Double>();
                    ArrayList<Double> requestCustomers = new ArrayList<Double>();

                    ArrayList<Integer> idDepots = new ArrayList<Integer>();
                    ArrayList<Double> axisXDepots = new ArrayList<Double>();
                    ArrayList<Double> axisYDepots = new ArrayList<Double>();
                    ArrayList<ArrayList<Integer>> countVehicles = new ArrayList<ArrayList<Integer>>();
                    ArrayList<ArrayList<Double>> capacityVehicles = new ArrayList<ArrayList<Double>>();

                    ArrayList<ArrayList<Double>> listDistances = new ArrayList<ArrayList<Double>>();

                    loadFile.loadCountVehiclesForDepot(countVehicles);
                    loadFile.loadCapacityVehiclesForHFVRP(capacityVehicles);
                    loadFile.loadCustomers(idCustomers, axisXCustomers, axisYCustomers, requestCustomers);
                    loadFile.loadDepots(idDepots, axisXDepots, axisYDepots);

                    loadFile.fillListDistances(idCustomers, axisXCustomers, axisYCustomers, idDepots, axisXDepots, axisYDepots, listDistances);

                    ProblemType typeProblem = ProblemType.HFVRP;
                    OrderType orderType = OrderType.Descending;
                    HeuristicType heuristicType = HeuristicType.NearestNeighborWithRLC;

                    if(StrategyHeuristic.getStrategyHeuristic().loadHFVRP(idCustomers, requestCustomers, idDepots, 
                            countVehicles.get(0), capacityVehicles.get(0), listDistances, 
                            axisXCustomers, axisYCustomers, axisXDepots, axisYDepots, typeProblem, orderType))
                    {
                                    StrategyHeuristic.getStrategyHeuristic().executeHeuristic(100, heuristicType);
                                    Solution result = StrategyHeuristic.getStrategyHeuristic().getBestSolution();
                                    double cost = StrategyHeuristic.getStrategyHeuristic().getTotalCostSolution();
                                    int requestByRoute = StrategyHeuristic.getStrategyHeuristic().getRequestByRoute().size();
                                    long time = StrategyHeuristic.getStrategyHeuristic().getTimeExecute();

                                    System.out.println(" ");
                                    System.out.println("------------------------------------------");
                                    //System.out.println("INSTANCIA: P" + (i + 1));
                                    System.out.println("CANTIDAD DE EJECUCIONES: " + 100);
                                    System.out.println("HEURÍSTICA DE CONSTRUCCIÓN: " + heuristicType);
                                    System.out.println("COSTO TOTAL: " + cost);
                                    System.out.println("TOTAL DE RUTAS: " + requestByRoute);
                                    System.out.println("TIEMPO DE EJECUCIÓN: " + time + " milisegundos");
                                    //System.out.println("TIEMPO DE EJECUCIÓN: " + time);
                                    System.out.println(" ");
                                    for(int j = 0; j < requestByRoute; j++)
                                        System.out.println("R" + (j+1) + result.getListRoutes().get(j).getListIdCustomers());
                                    System.out.println("------------------------------------------");
                            }
            printStream.close();
            fileOutputStream.close();

            System.setOut(System.out); // Restaurar la salida estándar
            } catch (IOException e) {
                e.printStackTrace();
            }
            //}
    }
    
}
